apps/recnet-api/src/utils/error/recnet.error.ts

```diff
@@ -0,0 +1,40 @@
+import { HttpStatus } from "@nestjs/common";
+
+import { ErrorCode, errorMessages } from "./recnet.error.const";
+
+export class RecnetError extends Error {
+  private errorCode: number;
+  private status: number;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  private extra: any | null = null;
+
+  constructor(
+    errorCode: number = ErrorCode.INTERNAL_SERVER_ERROR,
+    status: number = HttpStatus.INTERNAL_SERVER_ERROR,
+    message = "",
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    extra?: any
```

Comment:
What's the use of `extra` exactly?

Reply:
It provides additional information to the api consumers. For example, I wrapped the zod error message into the extra field.